mapper/thin_edge_json/src/serialize.rs

```diff
@@ -0,0 +1,118 @@
+use chrono::offset::FixedOffset;
+use chrono::DateTime;
+use std::io::Write;
+
+use crate::measurement::GroupedMeasurementVisitor;
+pub struct ThinEdgeJsonSerializer {
+    buffer: Vec<u8>,
+    is_within_group: bool,
+    needs_separator: bool,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ThinEdgeJsonSerializationError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error(transparent)]
+    MeasurementCollectorError(#[from] MeasurementStreamError),
+}
+#[derive(thiserror::Error, Debug)]
+pub enum MeasurementStreamError {
+    #[error("Unexpected time stamp within a group")]
+    UnexpectedTimestamp,
+
+    #[error("Unexpected end of data")]
+    UnexpectedEndOfData,
+
+    #[error("Unexpected end of group")]
+    UnexpectedEndOfGroup,
+
+    #[error("Unexpected start of group")]
+    UnexpectedStartOfGroup,
+}
+
+impl ThinEdgeJsonSerializer {
+    pub fn new() -> Self {
+        let mut serializer = ThinEdgeJsonSerializer {
+            buffer: Vec::new(),
+            is_within_group: false,
+            needs_separator: false,
+        };
+        serializer.buffer.push(b'{');
+        serializer
+    }
+
+    // pub fn start(&mut self) -> Result<(), ThinEdgeJsonSerializationError> {
+    //     self.buffer.push(b'{');
+    //     self.needs_separator = false;
+    //     Ok(())
+    // }
```

Comment:
No commented out code!

In that case I would uncomment this function and use it in the constructor. To better convey the intent.